arch/arc/core/mpu/arc_mpu.c

```diff
@@ -447,7 +445,17 @@ void arc_core_mpu_configure_user_context(struct k_thread *thread)
 	u32_t size = thread->stack_info.size;
 
 	/* for kernel threads, no need to configure user context */
-	if (!thread->arch.priv_stack_start) {
+	if (!(thread->base.user_options & K_USER)) {
+#if defined(CONFIG_APP_SHARED_MEM) && CONFIG_ARC_MPU_VER == 3
+/*
+ * APP_SHARED_MEM is handled here, all privileged threads have the right
+ * to access it. APPLICATION_MEMORY will be handled as a static memory region
+ */
+		base = (u32_t)&_app_smem_start;
+		size = (u32_t)&_app_smem_size;
+		_region_init(_get_region_index_by_type(THREAD_APP_DATA_REGION)
+		, base, size, _get_region_attr_by_type(THREAD_APP_DATA_REGION));
```

Comment:
please fix this indentation, the next line should line up with the opening parenthesis and the comma should not start the line